### DISCUSSION TITLE
MQTT States -> List of all states?

### DISCUSSION THREAD

**Discussion by ottelo9** [rand256/valetudo#410] [Q&A]
I want to send a push notification to my smartphone via Home Assistant in case of an error. MQTT already works with my HA. I dont now what I could use for that

Topic: valetudo/rockrobo/attributes -> states oder valetudo_state ?

For that I must know the state ID or whatever to send a message if my S50 is stucked.

**Comment by rand256** [comment]
.valetudo_states attribute corresponds to one of the states actually used in low-level device's firmware: https://github.com/rand256/valetudo/blob/5eb81dea5c758bc36133d8143419078d9600d354/lib/miio/Vacuum.js#L1110-L1129

.states attribute corresponds to how aforementioned device states are mapped to HA states (as there were very few states in HA compared to those actually used on the device at least at time MQTT client was coded). Below you can see the correlation for device_state <--> HA_state: https://github.com/rand256/valetudo/blob/5eb81dea5c758bc36133d8143419078d9600d354/lib/MqttClient.js#L61-L76
